docs/SNICKER.md

```diff
@@ -0,0 +1,322 @@
+
+The upshot - **make sure you actually generate wallet files for all wallets you're going to test SNICKER with**, otherwise you will not even see the created coins on the receiver side.
+
+Additionally, when you view the wallet with wallet-tool, you need to use `--recoversync`, as the default fast sync won't see imported keys. If you took these two steps, your tests should correctly show the post-SNICKER created coins.
+
+<a name="checking" />
+
+## Checking for SNICKER coinjoins
+
+SNICKER is logged to:
+
+`~/.joinmarket/logs/SNICKER/SNICKER-joinmarket-wallet-xxxxxx.log`
+
+i.e. there is one SNICKER log file for each wallet.
+This should show all transactions that were detected and broadcast.
+
+<a name="testing" />
+
+## A testing workflow
+
+This is a scenario for a single user, using either regtest or signet.
+
+Generate a minimum of two joinmarket wallets with `python wallet-tool.py generate`, as noted above, native (or at least, both the same type).
+
+Fund them both. The receiver needs at least two utxos to create the seed transaction.
+
+Create a seed fake-SNICKER transaction in the receiver wallet, using the script `snicker-seed-tx.py`.
+
+Start the test server. Navigate to `scripts/snicker` and run `python snicker-server.py` - no arguments should be needed, and this will generate an onion running serving on port 80; the onion hostname is displayed:
+
+```
+(jmvenv) waxwing@here~/testjminstall/joinmarket-clientserver/scripts/snicker$ python snicker-server.py 
+User data location: 
+Attempting to start onion service on port: 80 ...
+Your hidden service is available: 
+xpkqk2cy2h2ay5iecwcod5ka36nxj2tsiyczk2w5c6o7h5g57w3xg4id.onion
+```
+
+This is ephemeral, obviously we intend the real servers to be long-running. The one in the default config should exist and be long-running already, but of course your tests don't need to rely on this. Add:
+
+```
+
+[SNICKER]
+enabled = true
+servers= xpkqk2cy2h2ay5iecwcod5ka36nxj2tsiyczk2w5c6o7h5g57w3xg4id.onion,
+```
+... to a `joinmarket.cfg` that you add inside `scripts/snicker`, by copying it from `scripts/` or wherever you keep your testing `joinmarket.cfg` file. (This manual annoyance is part of testing, it won't be needed in mainnet usage of course ... alternatively just put a signet/regtest custom `joinmarket.cfg` in your `.joinmarket` folder, but this is hardly less annoying).
+
+`servers=` there is a comma separated list.
+
+You're now ready to do the two steps: (a) create a proposal and upload it, (b) download proposals (as the receiver identity/wallet) and complete coinjoins. It could be different people doing (a) and (b) of course but here we're assuming one tester doing everything (see two wallets above).
```

Comment:
Here a markdown list could be nice and clear. 

* a) Create a proposal and upload it.
* b) Download proposals (as the receiver identity/wallet) and complete coinjoins.

It could be different people doing (a) and (b) of course but here we're assuming one tester doing everything (see two wallets above).

Reply:
Not needed